### PR TITLE
[#102957884] Support certificate authentication for service principal

### PR DESCRIPTION
**Pull request by hovsepm**

### CODE REVIEW

**Comment by hovsepm** [comment]
on-demand job - http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/288/

File: src/ResourceManager/Profile/Commands.Profile/Microsoft.Azure.Commands.Profile.dll-Help.xml [hunk 1]
```diff
@@ -365,7 +365,7 @@ Add-AzureRmAccount -Credential $credential -TenantId xxxx-xxxx-xxxx-xxxx -Servic
           <maml:para>Account: xxxx-xxxx-xxxx-xxxx
```
Comment (markcowl) [inline src/ResourceManager/Profile/Commands.Profile/Microsoft.Azure.Commands.Profile.dll-Help.xml]: Need to add help for new Add-AzureRmAccount parameter sets

**Comment by markcowl** [comment]
A couple of comments, otherwise LGTM

**Comment by hovsepm** [comment]
on-demand job - http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/293/

**Comment by markcowl** [comment]
@hovsepm  You need to change PSAzureAccount to have a CertificateThumbprint property and change the type converters AzureAccvount <=> PSAzureAccount

**Comment by markcowl** [comment]
@hovsepm the reason is this:  if you save your context out to a variable or use it in a pipeline, the CertificateThumbprint will be lost.

**Comment by hovsepm** [comment]
on-demand job - http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/296/

**Comment by markcowl** [comment]
LGTM

**Comment by markcowl** [comment]
@hovsepm failures

**Comment by hovsepm** [comment]
on-demand job - http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/298/

**Comment by hovsepm** [comment]
on-demand job - http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/302/

**Pull request merged**